controllers/remote/cluster.go

```diff
@@ -26,6 +27,11 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const (
+	defaultClientTimeout   = 10 * time.Second
+	remoteClusterUserAgent = "remote-cluster"
```

Comment:
We might want to build something like https://github.com/kubernetes/client-go/blob/9761a13537eb881bb91c13819ba223eda84ac0af/rest/config.go#L463-L477

but specific to Cluster API?

Reply:
So instead of something like `kubernetes/` we could have `cluster.x-k8s.io/` for example?

Reply:
ok cool, I can do that

Reply:
+1, it might be good to also have a component that is defined at initialization time. That way we could differentiate which  component is using the remote client, which could potentially be different controllers or even different controller managers